### DISCUSSION TITLE
How to disable window autofocus?

### DISCUSSION THREAD

**Discussion by mrBars1k** [hyprwm/Hyprland#8660] [Q&A] [answered]
When I launch steam or get a call in telegram, the mouse cursor and focus immediately move to these windows and the workspaces on which these windows are located.

I can do things in the browser and want to left-click, but the cursor moves to telegram and I can accidentally click on something in it.

Or I want to set steam to open in the background while I do other things, but steam takes focus twice during startup and I have to wait.

Is it possible to remove autofocus capability from apps?

**Comment by fufexan** [comment]
Set `misc:focus_on_activate` to `false` (default). https://wiki.hyprland.org/Configuring/Variables/#misc

**Reply by mrBars1k** [reply]
Yeah, I did that, but it doesn't take away steam's ability to steal focus

**Comment by UrsusMortiferum** [comment, marked as answer]
This is what helped me: https://github.com/hyprwm/Hyprland/issues/5412.

Sample for signal, but works for any given class
`windowrulev2 = workspace 3 silent, class:^Signal$`

Unfortunately, it seems that `nofocus` or earlier suggested `focus_on_activate` do not work.

**Reply by mrBars1k** [reply]
Thank you! It really works
